deploy/autoscaler/member-operator-autoscaler.yaml

```diff
@@ -0,0 +1,48 @@
+kind: Template
+apiVersion: template.openshift.io/v1
+metadata:
+  name: member-operator-autoscaler
+objects:
+  - apiVersion: scheduling.k8s.io/v1
+    kind: PriorityClass
+    metadata:
+      name: member-operator-autoscaling-buffer
+      labels:
+        toolchain.dev.openshift.com/provider: codeready-toolchain
+    value: -100
+    globalDefault: false
+    description: "This priority class is to be used by the autoscaling buffer pod only"
+  - apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+      name: autoscaling-buffer
+      namespace: ${NAMESPACE}
+      labels:
+        app: autoscaling-buffer
+        toolchain.dev.openshift.com/provider: codeready-toolchain
+    spec:
+      replicas: 1
```

Comment:
I was thinking about replicas too but decided to take care later of it. Regarding a smaller ratio.. What I tried to achieve here is to reserve entire node(s). But reserving smaller chunks of memory blocks (not necessarily holding the entire nodes "ready-to-be-used") is probably a better idea. In this case we could reserve N "user" blocks. Where each block represents an amount of memory one user could claim rather than reserving % of node memory.

I'm going to drop the ratio thing and introduce hardcoded memory requests & replicas. So, we could deploy a buffer with, let's say 8Gi per replica and 3 replicas total. Which would guarantee enough resources for at least 3 users (in the case all the users use everything they can).

Reply:
Done.

Reply:
Yes, this is much simpler and more straightforward :+1: